sootty/storage/wire.py

```diff
@@ -23,7 +23,12 @@ def __setitem__(self, key, value):
         self._data[key] = value
 
     def __getitem__(self, key):
-        return self._data.get(key)
+        if isinstance(key, slice):
+            return Wire(f"{self.name}[{key.start}:{key.stop}]", width=key.stop - key.start)._data.map(lambda x: x[key])
```

Comment:
The value of `width` is the number of bits of the wire, not the number of clock ticks (that would be length). I would add a `slice` method in valuechange.py and call it from here.